packages/main/src/TableSelection.ts

```diff
@@ -204,6 +217,181 @@ class TableSelection extends UI5Element implements ITableFeature {
+			const startIndex = this._table?.rows.indexOf(startRow);
+			const endIndex = this._table?.rows.indexOf(row);
+
+			if (startIndex === -1 || endIndex === -1 || startIndex === undefined || endIndex === undefined
+				|| row.key === startRow.key || row.key === this._rangeSelection.rows[this._rangeSelection.rows.length - 1].key) {
+				return;
+			}
+
+			const change = endIndex - startIndex;
+			this._handleRangeSelection(row, change);
+		} else if (row) {
+			this.informSelectionChange(row);
+			this._startRangeSelection(row, true);
+		}
+	}
+
+	/**
+	 * Start the range selection and initialises the range selection state
+	 * @param row starting row
+	 * @private
+	 */
+	_startRangeSelection(row: TableRow, isMouse = false) {
+		const state = this.isSelected(row);
+		if (isMouse && !state) {
+			// Do not initiate range selection if the row is not selected
+			return;
+		}
+
+		this._rangeSelection = {
+			state,
+			isUp: null,
+			rows: [row],
+			isMouse,
+		};
+	}
+
+	/**
+	 * Handles the range selection
+	 * @param targetRow row that is currently focused
+	 * @param change indicates direction
+	 * @private
+	 */
+	_handleRangeSelection(targetRow: TableRow, change: number) {
+		if (!this._rangeSelection) {
+			return;
+		}
+
+		const isUp = change > 0;
+		let selectionChanged = false;
+		this._rangeSelection.isUp ??= isUp;
+		if (isUp !== this._rangeSelection.isUp && !this._rangeSelection.isMouse) {
+			// Only reverse selection for keyboard
+			if (this.isSelected(targetRow)) {
+				selectionChanged = true;
+			}
+			// Changing direction "reverse" the selection
+			this._reverseRangeSelection();
+		} else {
+			const endIndex = this._table!.rows.indexOf(targetRow);
+			const startIndex = endIndex - change;
+
+			this._table?.rows.slice(Math.min(startIndex, endIndex), Math.max(startIndex, endIndex) + 1).forEach(row => {
+				if (!this._rangeSelection?.rows.includes(row)) {
+					this._rangeSelection?.rows.push(row);
+				}
+
+				if (this.isSelected(row) !== this._rangeSelection!.state) {
+					selectionChanged = true;
+				}
+
+				// Workaround required due to bug with mouse, where the checkbox stays unchecked, but the row is selected.
+				// Even invalidation does not change the state of the checkbox. Maybe an issue with lit's 'intelligent' rendering?
+				this._rangeSelection?.isMouse && row.shadowRoot?.querySelector("#selection-component")?.setAttribute("checked", "");
+				this._selectRow(row, this._rangeSelection!.state);
+			});
+		}
+
+		selectionChanged && this._fireEvent("change");
+	}
+
+	_stopRangeSelection() {
+		this._rangeSelection = null;
+	}
+
+	_reverseRangeSelection() {
+		const row = this._rangeSelection?.rows.pop();
+		if (row) {
+			this._selectRow(row, false);
+		}
+
+		if (this._rangeSelection?.rows.length === 1) {
+			this._rangeSelection.isUp = null;
+		}
+	}
+
+	_isSelectionCheckbox(e: Event) {
+		return e.composedPath().some((el: EventTarget) => (el as HTMLElement).id === "selection-component");
```

Comment:
anything might have an id with selection-component

Reply:
Tried to use a unique attribute instead (data-ui5-table-selection-component)

Reply:
Feel free to suggest any other ideas